src/components/JsPsychExperiment.jsx

```diff
@@ -1,19 +1,79 @@
-import React from "react";
-import { Experiment } from "jspsych-react";
-import { tl } from "../timelines/main";
+import React, { useEffect, useRef, useMemo } from "react";
+import { initJsPsych } from 'jspsych'
+import { buildTimeline } from "../timelines/main";
 
-function JsPsychExperiment({ dataUpdateFunction, dataFinishFunction }) {
+function JsPsychExperiment({
+  participantId,
+  studyId,
+  startDate,
+  taskVersion,
+  dataUpdateFunction,
+  dataFinishFunction,
+  height = "100%",
+  width = "100%"
+}) {
+  // This will be the div in the dom that holds the experiment.
+  // We reference it explicitly here so we can do some plumbing with react, jspsych, and events.
+  const experimentDivId = 'experimentWindow';
+  const experimentDiv = useRef(null);
+
+  const jsPsychOpts = {
+    display_element: experimentDivId,
+    on_data_update: (data) => dataUpdateFunction(data),
+    on_finish: (data) => dataFinishFunction(data),
+  };
+
+  // Create the instance of jsPsych that we'll reuse within the scoope of this JsPsychExperiment component.
+  // As of jspsych 7, we create our own jspsych instance(s) where needed instead of importing one global instance.
+  const setUpJsPsych = () => {
+    const jsPsych = initJsPsych(jsPsychOpts)
+    jsPsych.data.addProperties({
+      participant_id: participantId,
+      study_id: studyId,
+      start_date: startDate,
+      task_version: taskVersion
+    })
+    return jsPsych
+  }
+  const jsPsych = useMemo(setUpJsPsych, [participantId, studyId, startDate, taskVersion]);
+
+  // Build our jspsych experiment timeline (in this case a Honeycomb demo, you could substitute your own here).
+  const timeline = buildTimeline(jsPsych)
```

Comment:
I refactored a few utils like this one to accept the current, scoped instance of `jspsych` instead of having those utils import the global `jspsych`.